airflow/serialization/base_serialization.py

```diff
@@ -216,7 +216,11 @@ def _deserialize(cls, encoded_var):  # pylint: disable=too-many-return-statement
             return [cls._deserialize(v) for v in encoded_var]
 
         assert isinstance(encoded_var, dict)
-        var = encoded_var[Encoding.VAR]
+        try:
+            var = encoded_var[Encoding.VAR]
+        except KeyError as e:
+            logging.info("encoded_var: %s", encoded_var)
```

Comment:
Is this just debugging?

Reply:
Yes but I kept it intentionally as it would be useful to log what key was not found.

Happy to remove it if you think